### PR TITLE
vector-symbology-editor

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**

### CODE REVIEW

**Comment by ChristianBeilschmidt** [comment]
[Vektor-Symbology-Editor anpassen](https://trello.com/c/Yvf98yxv/40-vektor-symbology-editor-anpassen)

**Comment by michaelmattig** [comment]
I noticed something I don't get yet:
- Add ports layer to the map
- set fill color to derived from natlscale, linear gradient
- add breakpoint
- set first breakpoint value to 1, seconds to 100, first color red, second color green => all points are green
- change first breakpoint value to 5 => some points are red others are green

why is this?

**Comment by michaelmattig** [comment]
If I put labels to points, e.g. "name" for the ports data set, the map becomes laggy. we should investigate this

**Review by michaelmattig — CHANGES_REQUESTED** [review]
I don't like the instanceof's but it works quite nice. I have some comments still.

File: projects/wave-core-new/src/lib/layers/symbology/color-param-editor/color-param-editor.component.html [hunk 1]
```diff
@@ -0,0 +1,37 @@
+<wave-color-attribute-input
```
[the remark this answers is not in the document]
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/layers/symbology/color-param-editor/color-param-editor.component.html:1]: It was basically a copy of the previous design and you can just use one component for specifying that value or specifying something for a text value (which turned out not to be possible currently - compared to the old design).

File: projects/wave-core-new/src/lib/layers/symbology/vector-symbology-editor/vector-symbology-editor.component.ts [hunk 2]
```diff
@@ -0,0 +1,420 @@
...
+
+        this.initializeAttributes();
+    }
+
+    ngAfterViewInit(): void {}
+
+    ngOnDestroy(): void {}
+
+    get fillColor(): ColorParam {
+        if (this.symbology instanceof PointSymbology) {
```
[the remark this answers is not in the document]
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/layers/symbology/vector-symbology-editor/vector-symbology-editor.component.ts:77]: we said we will do this later :tm: or when we need to touch the component again

File: projects/wave-core-new/src/lib/layers/symbology/vector-symbology-editor/vector-symbology-editor.component.html [hunk 3]
```diff
@@ -0,0 +1,113 @@
+<wave-sidenav-header>Symbology Editor</wave-sidenav-header>
```
[the remark this answers is not in the document]
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/layers/symbology/vector-symbology-editor/vector-symbology-editor.component.html]: d1d5268

File: projects/wave-core-new/src/lib/layers/symbology/number-param-editor/number-param-editor.component.html [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    <mat-select placeholder="Attribute" [(ngModel)]="numberAttributeName">
+        <mat-option [value]="undefined"><em>None</em></mat-option>
+        <mat-option *ngFor="let attribute of attributes" [value]="attribute">{{ attribute }}</mat-option>
+    </mat-select>
+</mat-form-field>
+
+<ng-template [ngIf]="isDerived">
+    <mat-form-field appearance="fill">
+        <mat-label>Factor</mat-label>
+        <input matInput type="number" [(ngModel)]="factor" />
```
[the remark this answers is not in the document]
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/layers/symbology/number-param-editor/number-param-editor.component.html:17]: 924f212

File: projects/wave-core-new/src/lib/colors/color-breakpoint-input/color-breakpoint-input.component.ts [hunk 5]
```diff
@@ -79,27 +96,28 @@ export class ColorBreakpointInputComponent implements ControlValueAccessor, Afte
...
     }
 
     registerOnTouched(fn: () => void): void {
         this.onTouched = fn;
     }
 
     private propagateChange(): void {
-        if (this.onChange && this.colorBreakpoint) {
-            this.onChange(this.colorBreakpoint.clone());
+        if (this.colorBreakpoint) {
```
[the remark this answers is not in the document]
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/colors/color-breakpoint-input/color-breakpoint-input.component.ts:119]: 94d3cfb

**Pull request merged**